### DISCUSSION TITLE
Unauthorized error in creating web viewer app

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#3356] [Q&A] [answered]
Hi there,
I am upgrading my existing app to 3.0 but continuously getting "_The user is unauthorized. Please provide valid authentication credentials"_ error. In order to seperate out this issue i created a new test app using template- `npx create-react-app web_viewer_3.0 --template @itwin/web-viewer --scripts-version @bentley/react-scripts`. I also followed [this tutorial](https://www.itwinjs.org/learning/tutorials/develop-web-viewer/#developing-a-web-viewer) to generate new clientId, scopes and all but the same error is coming every time.
![image](https://user-images.githubusercontent.com/27068008/158313703-0273ab4b-260f-4cbc-83c1-1d045cb89678.png)

Here is the scopes generated while registering spa app in developer.bentley.com
![image](https://user-images.githubusercontent.com/27068008/158315544-59578030-5a61-4819-9fbc-469946110d73.png)

Can anyone point me in right direction?
Thanks!

**Comment by kmurari103** [comment, marked as answer]
Thanks,
I saw similar issue [here](https://github.com/iTwin/itwinjs-core/discussions/3160#discussioncomment-2104773) which says about updating the scopes to `imodelaccess:read imodels:read realitydata:read`. 
It solved the unauthorization issue.

**Comment by kmurari103** [comment]
I believe that tutorial needs to get updated. Also terminologies are updated which does not reflect in there _(like IMJS_CONTEXT_ID modified to IMJS_ITWIN_ID)_

I do have a doubt regarding the urls-
So for REDIRECT uris- i had something `"https://localhost:3000/signin-callback"` starting with `https `but on upgrading to 3.0, it does not work. Can't i use https for uris?

**Reply by aruniverse** [reply]
Thanks for bringing this up, we'll update the tutorial. Regarding https, I don't believe that works for localhost in development, but it should when you have a production build pointing to an actual host
